pkg/kubeapiserver/server/insecure_handler.go

```diff
@@ -38,8 +38,9 @@ func BuildInsecureHandlerChain(apiHandler http.Handler, c *server.Config) http.H
 	handler = genericapifilters.WithAudit(handler, c.AuditBackend, c.AuditPolicyChecker, c.LongRunningFunc)
 	handler = genericapifilters.WithAuthentication(handler, server.InsecureSuperuser{}, nil, nil)
 	handler = genericfilters.WithCORS(handler, c.CorsAllowedOriginList, nil, nil, nil, "true")
-	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.LongRunningFunc, c.RequestTimeout)
+	handler = genericfilters.WithTimeoutForNonLongRunningRequests(handler, c.LongRunningFunc)
```

Comment:
can I get a comment here explaining why I have a second timeout filter here?

Reply:
@deads2k `WithTimeoutForNonLongRunningRequests` is a filter that times out a request after `60s` if the response hasn't been written yet. The new filter `WithRequestDeadline` sets the context deadline.  

I guess we still need the `WithTimeoutForNonLongRunningRequests` if processing not bound to any context goes rogue. in this case we will send a timeout error to the caller after `60s`. @sttts any thoughts here?

Reply:
`WithTimeoutForNonLongRunningRequests` has that support: it calls the rest of the request handling in a go-routine with the context with deadline. The go-routine can keep running, while the timeout logic will return a timeout to the client.

Reply:
Can I get these comments as comments in the code for future david?